graphium/trainer/losses.py

```diff
@@ -63,14 +63,14 @@ def forward(self, input: Tensor, target: Tensor, nan_targets: Tensor = None) ->
 
         # set input and target with nans to 0s for regression loss
         if nan_targets is not None:
-            input[nan_targets] = 0.0
-            target[nan_targets] = 0.0
+            input = torch.masked_fill(input, nan_targets.unsqueeze(1), 0)
```

Comment:
These changes were necessary to enable gradient computation on my side (the previous, in-place modification caused this tedious error: `one of the variables needed for gradient computation has been modified by an in-place operation`). Not sure why the tests aren't catching this or whether this is just an issue on my side. @DomInvivo your call whether we apply this change, otherwise I can also remove it. Will definitely not be as fast and memory-friendly as the in-place but I also cannot estimate how impactful it really is to change to out-of-place (essentially creates a copy but since I am assigning it to the same variable, the garbage collector should remove the old tensor)